### PR TITLE
Add birthdays

### LINKED ISSUE #12: All birthdays this week

### PR DESCRIPTION
**Pull request by DavidKang**
close #12

### CODE REVIEW

File: app/controllers/dashboards_controller.rb [hunk 1]
```diff
@@ -8,6 +8,8 @@ def index
     @public_holidays = Holidays.between(Time.zone.today.beginning_of_week,
                                         Time.zone.today.end_of_week,
                                         %i[es gb cz])
+    @birthdays = User.find_birthdays_for(Time.zone.today.beginning_of_week,
```
Comment (Ana06) [inline app/controllers/dashboards_controller.rb]: that will also render a bithday that was on monday on wednesday, right? why?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def github_link_to(login)
   def avatar(user, size: 200, gravatar: false)
     klass = gravatar ? 'ui avatar image mini' : 'ui fluid circular image'
 
-    url = if user.github_login
+    url = if user.github_login.present?
```
Comment (Ana06) [inline app/helpers/application_helper.rb]: how is that related to birthdays?

File: Gemfile [hunk 3]
```diff
@@ -43,6 +43,7 @@ gem 'country_select'
 gem 'flag-icon-sass'
 
 gem 'identicon'
+gem 'birthday', '~> 0.3.0'
```
Comment (Ana06) [inline Gemfile]: do we really need a gem for that? it is nothing we couldn't implement ourselves. It is easy, we would have more independence and we don't have the gem dependency.

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (DavidKang) [reply app/helpers/application_helper.rb]: right!

File: app/models/user.rb [hunk 4]
```diff
@@ -14,6 +14,12 @@ class User < ApplicationRecord
       .where('vacations.start_date <= :today AND vacations.end_date >= :today', today: Time.zone.today)
   end)
 
+  def self.birthdays_of_this_week
+    select do |user|
+      user.birthday? && user.birthday.change(year: Time.zone.now.year).in?((Time.zone.today..Time.zone.today + 6).to_a)
```
Comment (Ana06) [inline app/models/user.rb]: you can use `cover`, so you don't need to convert the range to an array :wink:

`(Time.zone.today..Time.zone.today + 6).cover? user.birthday.change(year: Time.zone.now.year)`

**Comment by DavidKang** [comment]
@Ana06, please re-review it please :smile_cat:

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -10,4 +10,14 @@
   context '#fullname' do
     it { expect(user.fullname).to eq("#{user.name} #{user.surname}") }
   end
+
+  context '#self.birthdays_of_this_week' do
+    let!(:user1) { create(:user, birthday: 20.years.ago) }
```
Comment (Ana06) [inline spec/models/user_spec.rb]: I would use property tests for this, but we can merge it and I'll do it afterwards :wink:
